sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py

```diff
@@ -0,0 +1,210 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from logging import getLogger
+from typing import Any, Dict
+
+from azure.monitor.opentelemetry._constants import (
+    DISABLE_LOGGING_ARG,
+    DISABLE_METRICS_ARG,
+    DISABLE_TRACING_ARG,
+    EXCLUDE_INSTRUMENTATIONS_ARG,
+    INSTRUMENTATION_CONFIG_ARG,
+    LOGGER_NAME_ARG,
+    LOGGING_EXPORT_INTERVAL_MS_ARG,
+    LOGGING_LEVEL_ARG,
+    METRIC_READERS_ARG,
+    RESOURCE_ARG,
+    SAMPLING_RATIO_ARG,
+    TRACING_EXPORT_INTERVAL_MS_ARG,
+    VIEWS_ARG,
+)
+from azure.monitor.opentelemetry._types import ConfigurationValue
+from azure.monitor.opentelemetry.exporter import (
+    ApplicationInsightsSampler,
+    AzureMonitorLogExporter,
+    AzureMonitorMetricExporter,
+    AzureMonitorTraceExporter,
+)
+from azure.monitor.opentelemetry.util.configurations import _get_configurations
+from opentelemetry._logs import get_logger_provider, set_logger_provider
+from opentelemetry.instrumentation.dependencies import (
+    get_dist_dependency_conflicts,
+)
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.metrics import set_meter_provider
+from opentelemetry.sdk._logs import LoggerProvider, LoggingHandler
+from opentelemetry.sdk._logs.export import BatchLogRecordProcessor
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics.export import PeriodicExportingMetricReader
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from opentelemetry.trace import get_tracer_provider, set_tracer_provider
+from pkg_resources import iter_entry_points
+
+_logger = getLogger(__name__)
+
+
+_SUPPORTED_INSTRUMENTED_LIBRARIES = (
+    "django",
+    "fastapi",
+    "flask",
+    "psycopg2",
+    "requests",
+    "urllib",
+    "urllib3",
+)
+
+
+InstrumentationConfig = Dict[str, Dict[str, Any]]
+
+
+def configure_azure_monitor(**kwargs) -> None:
+    """
+    This function works as a configuration layer that allows the
+    end user to configure OpenTelemetry and Azure monitor components. The
+    configuration can be done via arguments passed to this function.
+    :keyword str connection_string: Connection string for your Application Insights resource.
+    :keyword Sequence[str] exclude_instrumentations: Specifies instrumentations you want to disable.
```

Comment:
Our priority with the distro was to simplify the more intensive setup that was required by the distro. So enabling the most automatically is beneficial. As for making the instrumentations dependencies, that seems important for stability and officially supporting certain instrumentations. Customers can of course add additional instrumentations and trigger them themselves outside of configure_azure_monitor.

Reply:
One of the main purposes of the distro is to be a "family pack" whereby customers could download a set of our officially supported instrumentations. If we want stability, don't we need to not only have the instrumentations as requirements, but vendor them too? I think I would be open to the opt in approach, but product recently had me remove even the exclude-instrumentations parameter.